crypto/signer.go

```diff
@@ -43,3 +56,42 @@ func (s Signer) Sign(data []byte) (trillian.DigitallySigned, error) {
 		HashAlgorithm:      s.hasher.HashAlgorithm(),
 		Signature:          sig}, nil
 }
+
+// LogRootToJson returns the JSON representation of a signed log root that should be fed to
+// objecthash when verifying a signed root.
+// TODO(Martin2112): Decouple code to verify roots from Signer as you need a private
+// key to create these objects.
+func (s Signer) LogRootToJson(root trillian.SignedLogRoot) (string, error) {
```

Comment:
Do you need to do this round-trip via JSON?
If this returned a `map[string]interface{}` with appropriate keys set (which is what I think you'll get with the JSON round-trip) then you could call `objecthash.ObjectHash()` directly.